c++/src/kj/compat/tls.c++

```diff
@@ -248,16 +243,26 @@ public:
   }
 
   kj::Promise<size_t> tryRead(void* buffer, size_t minBytes, size_t maxBytes) override {
-    return tryReadInternal(buffer, minBytes, maxBytes, 0);
+    size_t alreadyDone = 0;
+    while (true) {
+      auto n = co_await sslCall([this,buffer,maxBytes]() { return SSL_read(ssl, buffer, maxBytes); });
+      if (n >= minBytes || n == 0) {
+        co_return alreadyDone + n;
+      } else {
+        buffer = reinterpret_cast<byte*>(buffer) + n;
+        minBytes -= n;
+        maxBytes -= n;
+        alreadyDone += n;
+      }
+    }
   }
 
   Promise<void> write(const void* buffer, size_t size) override {
-    return writeInternal(kj::arrayPtr(reinterpret_cast<const byte*>(buffer), size), nullptr);
+    return writeInternal({kj::arrayPtr(reinterpret_cast<const byte*>(buffer), size)});
```

Comment:
This isn't safe -- you're passing a temporary array (the outer array) to a coroutine which expects the array to remain valid across co_awaits. This is why `writeInternal()` was structured the way it was before, with separate "first" and "rest" arrays. Can you keep the old approach?

Reply:
good point. Wouldn't `co_return co_await` solve this problem rather than have first/rest arguments?

Reply:
(I'm also wondering if we can setup asan build for capnproto?)

Reply:
`co_return co_await` would solve it at the expense of allocating an extra coroutine frame.